chainer/functions/vae.py

```diff
@@ -0,0 +1,106 @@
+import math
+
+import chainer.functions as F
+from chainer import variable
+
+
+def gaussian_kl_divergence(z_mean, z_ln_var):
+    """Calculate KL-divergence between a given gaussian and the normal one.
+
+    Given two variable ``z_mean`` representing :math:`\\mu` and ``z_ln_var``
+    representing :math:`\\log(\\sigma^2)`, this function returns a variable
+    representing KL-divergence between :math:`N(\\mu, \\sigma^2)` and
+    the normal Gaussian :math:`N(0, 1)`.
```

Comment:
We should note that `N(mu, sigma^2)` and `N(0, 1)` is multi-dimensional. In concrete, we should add that `mu` and sigma is vectors. We also write `N(mu, sigma^2)` and `N(0, 1)` as `N(mu, sigma^2 I)` and `N(0, I)` where I is a identity matrix.